lib/chef/provider/package/chocolatey.rb

```diff
@@ -238,6 +242,32 @@ def available_packages
           @available_packages
         end
 
+        # Getting packages from Artifactory server and return as candidate
+        def available_packages_from_artifactory
+          return @available_packages_from_artifactory if @available_packages_from_artifactory
+          @available_packages_from_artifactory = {}
+          package_name_array.length.times do |item|
```

Comment:
this looks like it should just be `package_name_array.each do |pkg|` and `list -r #{pkg}` below.